modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImplTest.java

```diff
@@ -997,6 +1005,49 @@ public void testTimeoutDdl() {
         awaitExecutionTimeout(execService, plan, 500, QueryCancelledException.class);
     }
 
+    @Test
+    public void testTimeoutKill() {
+        SqlOperationContext planCtx = operationContext(null).build();
+        QueryPlan plan = prepare("KILL QUERY '" + randomUUID() + '\'', planCtx);
+
+        assertInstanceOf(KillPlan.class, plan);
+
+        int deadlineMillis = 500;
+
+        ExecutionServiceImpl<?> execService = executionServices.get(0);
+
+        Duration delay = Duration.of(deadlineMillis * 4, ChronoUnit.MILLIS);
+
+        killHandlerRegistry.register(new OperationKillHandler() {
+            @Override
+            public CompletableFuture<Boolean> cancelAsync(String operationId) {
```

Comment:
I think we do not need `Thread.sleep` here and we can simply return a `CompletableFuture` that never completes.

Reply:
Done, thanks.